src/main/java/com/exasol/containers/ExasolContainer.java

```diff
@@ -99,24 +106,32 @@ public ExasolContainer() {
     // [impl->dsn~exasol-container-uses-privileged-mode~1]
     @Override
     protected void configure() {
-        exposePorts();
+        if (this.portAutodetectFailed) {
+            if (getExposedPorts().isEmpty()) {
+                throw new IllegalArgumentException(
+                        "Could not detect internal ports for custom image. Please specify the port explicitly using withExposedPorts().");
+            } else {
+                logger().warn("Could not detect internal ports for custom image. "
+                        + "Don't forget to expose the database and BucketFs ports yourself.");
+            }
+        }
+        logger().debug("Exposing ports: {}", this.getExposedPorts());
```

Comment:
Do you have to call `exposePorts()` here?

Reply:
it was a local method. Since I removed it's content, I also removed it from here